game-core/src/main/java/org/triplea/util/LocalizeHtml.java

```diff
@@ -1,58 +1,42 @@
 package org.triplea.util;
 
-import com.google.common.annotations.VisibleForTesting;
-import games.strategy.triplea.ResourceLoader;
-import games.strategy.triplea.ui.UiContext;
-import java.net.URL;
-import java.util.HashMap;
-import java.util.Map;
+import java.io.File;
+import java.nio.file.Path;
 import java.util.Optional;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import lombok.experimental.UtilityClass;
-import lombok.extern.slf4j.Slf4j;
+import org.triplea.io.FileUtils;
 
 /**
  * Provides methods that convert relative links within a game description into absolute links that
- * will work on the local system.
+ * will work on the local system. Links that are already absolute (those beginning with 'http') will
+ * not be changed.
  */
-@Slf4j
 @UtilityClass
 public final class LocalizeHtml {
-  private static final String ASSET_IMAGE_FOLDER = "doc/images/";
-  private static final String ASSET_IMAGE_NOT_FOUND = "notFound.png";
   // Match the <img> src
   private static final Pattern PATTERN_HTML_IMG_SRC_TAG =
       Pattern.compile(
           "(<img[^>]*src\\s*=\\s*)(?:\"([^\"]+)\"|'([^']+)')([^>]*/?>)", Pattern.CASE_INSENSITIVE);
 
   /**
-   * This is only useful once we are IN a game. Before we go into the game, resource loader will
-   * either be null, or be the last game's resource loader.
+   * Replaces relative image links within a given {@code htmlText} with absolute links that point to
+   * the correct location on the local file system.
    */
-  public static String localizeImgLinksInHtml(final String htmlText) {
-    return localizeImgLinksInHtml(htmlText, UiContext.getResourceLoader());
-  }
-
-  /**
-   * Replaces relative image links within the HTML document {@code htmlText} with absolute links
-   * that point to the correct location on the local file system.
-   */
-  public static String localizeImgLinksInHtml(final String htmlText, final String mapNameDir) {
-    return htmlText == null || mapNameDir == null || mapNameDir.isBlank()
-        ? htmlText
-        : localizeImgLinksInHtml(htmlText, new ResourceLoader(mapNameDir));
-  }
-
-  @VisibleForTesting
-  static String localizeImgLinksInHtml(final String htmlText, final ResourceLoader loader) {
+  public static String localizeImgLinksInHtml(final String htmlText, final File mapContentFolder) {
```

Comment:
So in case the resource does not exist we will no longer notice at this point, right?
And because the HTML is not really handled by us, this will result in a silent error, just like it would in a normal web browser

Reply:
The resource loader was only be used to get the map folder location, it was not actually loading the image or validating that it was there. The previous logic was doing everything as a String operation, so we were not noticing it either way.